checksuite/kv/errmap/check_errmap.go

```diff
@@ -0,0 +1,89 @@
+package errmap
+
+import (
+	"github.com/couchbase/gocbcore/v9/memd"
+	"github.com/couchbaselabs/gocaves/checks"
+	"github.com/couchbaselabs/gocaves/mock"
+	"log"
+	"sync/atomic"
+)
+
+// CheckErrMapLinearRetry confirms that the SDK can successfully use an unknown status code to perform a lookup in
+// the error map and retry accordingly.
+func CheckErrMapLinearRetry(t *checks.T) {
+	checkErrorMapRetry(t, 0x7ff0, "7ff0", mock.ErrorMapError{
+		Name:  "DUMMY_ERROR_RETRY_EXPONENTIAL",
+		Desc:  "Dummy retry error for exponential backoff",
+		Attrs: []string{"auto-retry", "temp"},
+		Retry: &mock.ErrorMapRetry{
+			Strategy:    "constant",
+			Interval:    10,
+			After:       50,
+			MaxDuration: 500,
+		},
+	})
+}
+
+// CheckErrMapConstantRetry confirms that the SDK can successfully use an unknown status code to perform a lookup in
+// the error map and retry accordingly.
+func CheckErrMapConstantRetry(t *checks.T) {
+	checkErrorMapRetry(t, 0x7ff1, "7ff1", mock.ErrorMapError{
+		Name:  "DUMMY_ERROR_RETRY_LINEAR",
+		Desc:  "Dummy retry error for linear backoff",
+		Attrs: []string{"auto-retry", "temp"},
+		Retry: &mock.ErrorMapRetry{
+			Strategy:    "linear",
+			Interval:    2,
+			After:       50,
+			MaxDuration: 500,
+			Ceil:        50,
+		},
+	})
+}
+
+// CheckErrMapExponentialRetry confirms that the SDK can successfully use an unknown status code to perform a lookup in
+// the error map and retry accordingly.
+func CheckErrMapExponentialRetry(t *checks.T) {
+	checkErrorMapRetry(t, 0x7ff2, "7ff2", mock.ErrorMapError{
+		Name:  "DUMMY_ERROR_RETRY_CONSTANT",
+		Desc:  "Dummy retry error for constant backoff",
+		Attrs: []string{"auto-retry", "temp"},
+		Retry: &mock.ErrorMapRetry{
+			Strategy:    "exponential",
+			Interval:    2,
+			After:       50,
+			MaxDuration: 500,
+			Ceil:        200,
+		},
+	})
+}
+
+func checkErrorMapRetry(t *checks.T, status memd.StatusCode, errMapStatus string, errMapErr mock.ErrorMapError) {
+	t.RequireFeature(checks.TestFeatureErrMap)
+	var i uint32
+	handler := func(source mock.KvClient, pak *memd.Packet, next func()) {
+		if atomic.AddUint32(&i, 1) > 3 {
+			next()
+			return
+		}
+
+		err := source.WritePacket(&memd.Packet{
+			Command: pak.Command,
+			Magic:   memd.CmdMagicRes,
+			Opaque:  pak.Opaque,
+			Status:  status,
+		})
+		if err != nil {
+			log.Printf("failed to write packet %+v to %+v", pak, source)
+		}
+	}
+
+	for _, n := range t.Mock().Nodes() {
+		n.ErrorMap().Extend(errMapStatus, errMapErr)
```

Comment:
I think it might be a better idea to have it so you can extend the error map and write a custom handler for the error map packet which returns the custom error map, but I'm not super confident on that.

Reply:
I'm unsure about this. That feels like it'll be harder to use and I can't imagine there'll be any usage outside of an Extend and a Clear that we'd actually need?